### ISSUE TITLE
AutoBogus fails to generate classes with WeakReference properties.

### ISSUE THREAD

**Issue by hultsk** [soenneker/soenneker.utils.autobogus#415]
We encountered an issue where AutoBogus fails when attempting to generate instances of a class containing private WeakReference properties. This behavior causes test engines to fail or hang after five or more iterations of instance generation.

## Steps to Reproduce:
Below is a minimal example that reproduces the issue:

``` c#
[TestClass]
public class DummyTests
{
    private class MyClass
    {
        private readonly WeakReference<MyClass> myWeakReference = new WeakReference<MyClass>(null);

        public string Name { get; set; }
    }

    [TestMethod]
    public void Test()
    {
        for (int i = 0; i < 6; i++)
        {
            AutoFaker.GenerateStatic<MyClass>();
        }
    }
}
```
## Observed Behavior:
The test engine fails or hangs on the sixth iteration when AutoFaker attempts to generate an instance of the class.

## Expected Behavior:
AutoBogus should generate instances of classes with private WeakReference properties without any errors, regardless of the number of iterations.

## Proposed Solution or Workaround:
While it is possible to work around this by implementing custom builders for affected classes, this adds complexity and makes AutoBogus less seamless for such cases. It would be helpful if AutoBogus could either handle WeakReference properties gracefully or document this limitation with guidance on how to mitigate it.

## Environment:

    Soenneker.Utils.AutoBogus version: 3.0.450
    .NET version: net8.0
    Test framework: MSTest.TestFramework 3.6.2
    
Additional Context:
This issue has practical implications when generating test data for classes relying on WeakReference for internal behaviors. It would be great to understand whether this is a fundamental limitation or an area that can be improved in AutoBogus.

**Comment by soenneker** [comment]
@hultsk I believe this is solved in edab4e9c2704f3cf2d2b5a5aeb56835aa89efff3

Let me know if the new version works for you, thanks.

**Comment by hultsk** [comment]
> @hultsk I believe this is solved in [edab4e9](https://github.com/soenneker/soenneker.utils.autobogus/commit/edab4e9c2704f3cf2d2b5a5aeb56835aa89efff3)
> 
> Let me know if the new version works for you, thanks.
> 
> I should mention `WeakReference` without a generic type parameter (i.e. declaration like `WeakReference _weakReference {get; set;}`) defaults the target object to type `string` (because there's no indication what it should be when AutoBogus is asked to generate it)
> 
> Your case here, with `MyClass` will generate a `MyClass` and set it on the `WeakReference`.

This works great, thank you for the quick response, and fast fix.
